app/controllers/pages_controller.rb

```diff
@@ -5,6 +5,10 @@ def home
   end
 
   def shortlist
-
+    @user = current_user #check shortlist for current user
+    @providers = Shortlist.all.map { |shortlist|
```

Comment:
it think line 9 should be changed to 
`@providers = @user.shortlists.map { *same as you had before* }`
because they way it is right now it will display all shortlisted providers of any users when i am not mistaking.